cpp/src/io/parquet/stats_filter_helpers.hpp

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "io/parquet/parquet_common.hpp"
+
+#include <cudf/ast/detail/expression_transformer.hpp>
+#include <cudf/ast/expressions.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/utilities/integer_utils.hpp>
+#include <cudf/detail/utilities/vector_factories.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/bit.hpp>
+#include <cudf/utilities/memory_resource.hpp>
+#include <cudf/utilities/span.hpp>
+#include <cudf/utilities/traits.hpp>
+
+#include <algorithm>
+#include <numeric>
+#include <string>
+
+namespace cudf::io::parquet::detail {
+
+/**
+ * @brief Base utilities for converting and casting stats values
+ *
+ * Derived classes handle row group or page-level statistics as needed.
+ *
+ */
+class stats_caster_base {
+ protected:
+  template <typename ToType, typename FromType>
+  static inline ToType targetType(FromType const value)
+  {
+    if constexpr (cudf::is_timestamp<ToType>()) {
+      return static_cast<ToType>(
+        typename ToType::duration{static_cast<typename ToType::rep>(value)});
+    } else if constexpr (std::is_same_v<ToType, string_view>) {
+      return ToType{nullptr, 0};
```

Comment:
What does this function do? Will the following suggestion work?
```suggestion
      return ToType{};
```

Reply:
The code is moved here but afaik, stats aren't helpful for `string_view` type cols so we just return an empty `string_view` here. @karthikeyann can you please confirm if my understanding is correct?

Reply:
I think the suggestion should work as well but let me check.

Reply:
Ok. Technically `{}` is _empty_ and `{nullptr,0}` is not really valid. 
Though it used in certain special conditions but I would like to minimize those when possible.